File: ai-chat-frontend/aichat-frontend/src/services/voiceService.js
import request from './api'

export const voiceService = {
  // 创建音色（管理员功能）
  createVoice(data) {
    return request.post('/voice/create', data)
  },

  // 获取所有音色（管理员功能）
  getAllVoices() {
    return request.get('/voice/list')
  },

  // 根据ID获取音色详情（管理员功能）
  getVoiceById(id) {
    return request.get(`/voice/${id}`)
  },

  // 更新音色（管理员功能）
  updateVoice(id, data) {
    return request.put(`/voice/${id}`, data)
  },

  // 删除音色（管理员功能）
  deleteVoice(id) {
    return request.delete(`/voice/${id}`)
  },

  // 启用/禁用音色（管理员功能）
  toggleVoiceStatus(id, enabled) {
    return request.put(`/voice/${id}/toggle`, null, {
      params: { enabled }
    })
  },

  // 查询所有启用的音色（用户功能）
  getEnabledVoices() {
    return request.get('/voice/enabled')
  },

  // 根据性别查询启用的音色（用户功能）
  getEnabledVoicesByGender(gender) {
    return request.get(`/voice/enabled/gender/${gender}`)
  },

  // 搜索音色（用户功能）
  searchVoices(keyword) {
    return request.get('/voice/search', {
      params: { keyword: keyword || '' }
    })
  }
}

File: ai-chat-web/ai-chat/src/main/java/com/k8/service/VoiceService.java
package com.k8.service;

import com.k8.dto.VoiceCreateDTO;
import com.k8.vo.VoiceQueryVO;
import com.k8.vo.VoiceVO;

import java.util.List;

/**
 * 音色服务接口
 * @Author: k8
 * @CreateTime: 2025-09-23
 * @Version: 1.0
 */
public interface VoiceService {
    
    /**
     * 创建音色（管理员功能）
     */
    VoiceVO createVoice(VoiceCreateDTO createDTO, String creatorId);
    
    /**
     * 获取所有音色（管理员功能）
     */
    List<VoiceVO> getAllVoices();
    
    /**
     * 根据ID获取音色详情（管理员功能）
     */
    VoiceVO getVoiceById(String id);
    
    /**
     * 更新音色（管理员功能）
     */
    VoiceVO updateVoice(String id, VoiceCreateDTO updateDTO);
    
    /**
     * 删除音色（管理员功能）
     */
    boolean deleteVoice(String id);
    
    /**
     * 启用/禁用音色（管理员功能）
     */
    boolean toggleVoiceStatus(String id, boolean enabled);
    
    /**
     * 查询所有启用的音色（用户功能）
     */
    List<VoiceQueryVO> getEnabledVoices();
    
    /**
     * 根据性别查询启用的音色（用户功能）
     */
    List<VoiceQueryVO> getEnabledVoicesByGender(String gender);
    
    /**
     * 搜索音色（用户功能）
     */
    List<VoiceQueryVO> searchVoices(String keyword);
}
